### PR TITLE
feat: project list UX and bulk actions

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/8077

### Description:

CleanShot 2024-11-06 at 14 52 29@2x

https://github.com/user-attachments/assets/8a092119-c429-4d2a-bb92-96ae9e18d6ba

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by TuringLovesDeathMetal** [comment]
@IrakliJani can we make the confirmation text just `delete` (i.e. `type 'delete' to confirm`)

This looks really great 😍 well done!

**Comment by IrakliJani** [comment]
@TuringLovesDeathMetal 
CleanShot 2024-11-06 at 17 25 53@2x

good?

File: packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx [hunk 1]
```diff
@@ -15,28 +32,68 @@ import { Can } from '../../common/Authorization';
...
-    const { mutate: updateActiveProjectMutation } =
+    const { mutateAsync: updateActiveProjectMutation } =
         useUpdateActiveProjectMutation();
 
+    const handleProjectSettingsClick = async () => {
+        if (!isCurrentProject) {
+            await updateActiveProjectMutation(project.projectUuid);
+        }
+        // a lot of cache invalidation happens after the mutation above
+        // so we need to wait for the next event loop to navigate to the new project
```
Comment (magnew) [inline packages/frontend/src/components/UserSettings/ProjectManagementPanel/index.tsx:65]: Nice

**Review by magnew — APPROVED** [review]
Very nice!
✅ Tested everything I could think of
✅ Code looks good!

**Pull request merged**